textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/dictionarymatcher/DictionaryMatcher.java

```diff
@@ -153,22 +167,40 @@ public ITuple getNextTuple() throws Exception {
     				return null;
     			}
     		}
-    		
-    		ITuple result = currentTuple;
-    		while (currentTuple != null) {
-    			result = matchTuple(currentDictionaryEntry, currentTuple);
-    			if (result != null) {
-    				advanceCursor();
+    		ITuple result = null;
+	    	while (currentTuple != null) {
+	    		result = matchTuple(currentDictionaryEntry, currentTuple);
+	    		if (result != null) {
+	    			advanceCursor();

```

Comment:
Create a new function called "computeNextMatchedTuple()", which includes the logic of moving the dictionary cursor.